learning_assistant/views.py

```diff
@@ -0,0 +1,101 @@
+"""
+V1 API Views
+"""
+import logging
+
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import status as http_status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+try:
+    from lms.djangoapps.courseware.access import get_user_role
+    from lms.djangoapps.courseware.toggles import learning_assistant_is_active
+    from common.djangoapps.course_modes.models import CourseMode
+    from common.djangoapps.student.models import CourseEnrollment
+except ImportError:
+    # If the waffle flag is false, the endpoint will force an early return.
+    learning_assistant_is_active = False
+
+from learning_assistant.models import CoursePrompt
+from learning_assistant.serializers import MessageSerializer
+from learning_assistant.utils import get_chat_response
+
+log = logging.getLogger(__name__)
+
+
+class CourseChatView(APIView):
+    """
+    View to retrieve chat response
+    """
+
+    authentication_classes = (SessionAuthentication, JwtAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, course_id):
+        """
+        Given a course ID, retrieve a chat response for that course.
+
+        Expected POST data: {
+            [
+                {'role': 'user', 'content': 'What is 2+2?'},
+                {'role': 'assistant', 'content': '4'}
+            ]
+        }
+
+        """
+
+        if not learning_assistant_is_active(course_id):
+            return Response(
+                status=http_status.HTTP_403_FORBIDDEN,
+                data={'detail': 'Learning assistant not enabled for course.'}
+            )
+
+        # get user does not have a verified enrollment, or is not staff, they should not have access
+        user_role = get_user_role(request.user, course_id)
+        enrollment_object = CourseEnrollment.get_enrollment(request.user, course_id)
+        enrollment_mode = enrollment_object.mode if enrollment_object else None
+        if (
+            (enrollment_mode not in CourseMode.VERIFIED_MODES)
+            and not user_role in ('staff', 'instructor')
+        ):
+            return Response(
+                status=http_status.HTTP_403_FORBIDDEN,
+                data={'detail': 'Must be staff or have valid enrollment.'}
+            )
+
+        prompt_text = CoursePrompt.get_prompt_by_course_id(course_id)
+        if not prompt_text:
+            return Response(
+                status=http_status.HTTP_404_NOT_FOUND,
+                data={'detail': 'Learning assistant not enabled for course.'}
+            )
+
+        message_list = request.data
```

Comment:
The request structure for messages is different than what is currently being done in https://github.com/kalebjdavenport/edx-chatbot/blob/c07f1f879e92488d606e86dfb40d1f9c2ff10d48/src/widgets/Xpert.jsx#L70-L74, but I think there is unnecessary nesting being done in the implementation.